### DISCUSSION TITLE
hardware transcoding for AV1 with OpenVino Intel Iris Xe does not work

### DISCUSSION THREAD

**Discussion by Str1atum** [immich-app/immich#9887] [Q&A]
Works for H265 in my setup but not for AV1 - seems like there is a driver missing:

> [Nest] 7  - 05/30/2024, 8:38:50 AM     LOG [ImmichMicroservices] [MediaService] Started encoding video 0f465c4c-934d-4e96-a8a0-3ca038d53a82 {"inputOptions":["-init_hw_device qsv=hw","-filter_hw_device hw"],"outputOptions":["-c:v av1_qsv","-c:a copy","-movflags faststart","-fps_mode passthrough","-map 0:1","-map 0:0","-bf 7","-refs 5","-g 256","-v verbose","-vf format=nv12,hwupload=extra_hw_frames=64,scale_qsv=-1:720","-preset 4","-global_quality 35","-maxrate 2500","-bufsize 5000"],"twoPass":false}
[Nest] 7  - 05/30/2024, 8:38:50 AM   ERROR [ImmichMicroservices] [MediaRepository] ffmpeg version 6.0.1-Jellyfin Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 12 (Debian 12.2.0-14)
  configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-ptx-compression --disable-static --disable-libxcb --disable-sdl2 --disable-xlib --enable-lto --enable-gpl --enable-version3 --enable-shared --enable-gmp --enable-gnutls --enable-chromaprint --enable-opencl --enable-libdrm --enable-libass --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-libbluray --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libopenmpt --enable-libdav1d --enable-libsvtav1 --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --enable-libfdk-aac --arch=amd64 --enable-libshaderc --enable-libplacebo --enable-vulkan --enable-vaapi --enable-amf --enable-libvpl --enable-ffnvcodec --enable-cuda --enable-cuda-llvm --enable-cuvid --enable-nvdec --enable-nvenc
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
[AVHWDeviceContext @ 0x290fc180340] Trying to use DRM render node for device 0, with matching kernel driver (i915).
[AVHWDeviceContext @ 0x290fc180340] libva: VA-API version 1.21.0
[AVHWDeviceContext @ 0x290fc180340] libva: User requested driver 'iHD'
[AVHWDeviceContext @ 0x290fc180340] libva: Trying to open /usr/lib/jellyfin-ffmpeg/lib/dri/iHD_drv_video.so
[AVHWDeviceContext @ 0x290fc180340] libva: Found init function __vaDriverInit_1_21
[AVHWDeviceContext @ 0x290fc180340] libva: va_openDriver() returns 0
[AVHWDeviceContext @ 0x290fc180340] Initialised VAAPI connection: version 1.21
[AVHWDeviceContext @ 0x290fc180340] VAAPI driver: Intel iHD driver for Intel(R) Gen Graphics - 24.1.5 (8068c2e).
[AVHWDeviceContext @ 0x290fc180340] Driver not found in known nonstandard list, using standard behaviour.
[AVHWDeviceContext @ 0x290fc180280] Use Intel(R) oneVPL to create MFX session, API version is 2.10, the required implementation version is 1.3
libva info: VA-API version 1.21.0
libva info: Trying to open /usr/lib/jellyfin-ffmpeg/lib/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_21
libva info: va_openDriver() returns 0
libva info: VA-API version 1.21.0
libva info: Trying to open /usr/lib/jellyfin-ffmpeg/lib/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_21
libva info: va_openDriver() returns 0
[AVHWDeviceContext @ 0x290fc180280] Initialize MFX session: implementation version is 2.10
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/diskstation-photos/PhotoLibrary/2024/05/Wilferdingen.mov':
  Metadata:
    major_brand     : qt  
    minor_version   : 0
    compatible_brands: qt  
    creation_time   : 2017-08-25T12:00:00.000000Z
  Duration: 00:01:18.67, start: 0.000000, bitrate: 20525 kb/s
  Stream #0:0[0x1](und): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 191 kb/s (default)
    Metadata:
      creation_time   : 2024-05-11T10:29:43.000000Z
      handler_name    : Core Media Audio
      vendor_id       : [0][0][0][0]
  Stream #0:1[0x2](und): Video: hevc (Main), 1 reference frame (hvc1 / 0x31637668), yuv420p(tv, bt709, left), 3840x2160, 20327 kb/s, 30 fps, 30 tbr, 600 tbn (default)
    Metadata:
      creation_time   : 2024-05-11T10:29:43.000000Z
      handler_name    : Core Media Video
      vendor_id       : [0][0][0][0]
      encoder         : HEVC
Stream mapping:
  Stream #0:1 -> #0:0 (hevc (native) -> av1 (av1_qsv))
  Stream #0:0 -> #0:1 (copy)
Press [q] to stop, [?] for help
[graph 0 input from stream 0:1 @ 0x290fc1f7bc0] w:3840 h:2160 pixfmt:yuv420p tb:1/600 fr:30/1 sar:0/1
[auto_scale_0 @ 0x290fc1f7e00] w:iw h:ih flags:'' interl:0
[Parsed_format_0 @ 0x290fc1f7980] auto-inserting filter 'auto_scale_0' between the filter 'graph 0 input from stream 0:1' and the filter 'Parsed_format_0'
[auto_scale_0 @ 0x290fc1f7e00] w:3840 h:2160 fmt:yuv420p sar:0/1 -> w:3840 h:2160 fmt:nv12 sar:0/1 flags:0x00000004
[AVHWDeviceContext @ 0x290fc184940] VAAPI driver: Intel iHD driver for Intel(R) Gen Graphics - 24.1.5 (8068c2e).
[AVHWDeviceContext @ 0x290fc184940] Driver not found in known nonstandard list, using standard behaviour.
[AVHWDeviceContext @ 0x290fc186180] VAAPI driver: Intel iHD driver for Intel(R) Gen Graphics - 24.1.5 (8068c2e).
[AVHWDeviceContext @ 0x290fc186180] Driver not found in known nonstandard list, using standard behaviour.
[Parsed_scale_qsv_2 @ 0x290fc1f7b00] Use Intel(R) oneVPL to create MFX session with the specified MFX loader
[Parsed_scale_qsv_2 @ 0x290fc1f7b00] VPP: input is video memory surface
[Parsed_scale_qsv_2 @ 0x290fc1f7b00] VPP: output is video memory surface
[auto_scale_0 @ 0x290fc1f7e00] w:3840 h:2160 fmt:yuv420p sar:0/1 -> w:3840 h:2160 fmt:nv12 sar:0/1 flags:0x00000004
    Last message repeated 2 times
[AVHWFramesContext @ 0x29117fd2d00] Use Intel(R) oneVPL to create MFX session, API version is 2.10, the required implementation version is 2.10
[AVHWFramesContext @ 0x29117fd2d00] Initialize MFX session: implementation version is 2.10
[av1_qsv @ 0x290fc212580] Using input frames context (format qsv) with av1_qsv encoder.
[av1_qsv @ 0x290fc212580] Encoder: input is video memory surface
[av1_qsv @ 0x290fc212580] Use Intel(R) oneVPL to create MFX session with the specified MFX loader
[av1_qsv @ 0x290fc212580] Using the constant quality with VBR algorithm (QVBR) ratecontrol method
[av1_qsv @ 0x290fc212580] Selected ratecontrol mode is unsupported
[av1_qsv @ 0x290fc212580] Current frame rate is unsupported
[av1_qsv @ 0x290fc212580] Current picture structure is unsupported
[av1_qsv @ 0x290fc212580] Current resolution is unsupported
[av1_qsv @ 0x290fc212580] Current pixel format is unsupported
[av1_qsv @ 0x290fc212580] some encoding parameters are not supported by the QSV runtime. Please double check the input parameters.
[vost#0:0/av1_qsv @ 0x290fc1d8b80] Error initializing output stream: Error while opening encoder for output stream #0:0 - maybe incorrect parameters such as bit_rate, rate, width or height
[AVIOContext @ 0x290fc205900] Statistics: 0 bytes written, 0 seeks, 0 writeouts
Terminating demuxer thread 0
[AVIOContext @ 0x290fc209f00] Statistics: 3239689 bytes read, 42 seeks
Conversion failed!
[Nest] 7  - 05/30/2024, 8:38:50 AM   ERROR [ImmichMicroservices] [MediaService] Error: ffmpeg exited with code 1: Conversion failed!
[Nest] 7  - 05/30/2024, 8:38:50 AM   ERROR [ImmichMicroservices] [MediaService] Error occurred during transcoding. Retrying with QSV acceleration disabled.

**Comment by mertalev** [comment]
What model is your processor?

**Comment by Str1atum** [comment]
13th Gen i9-13900H

**Reply by mertalev** [reply]
Your model only has AV1 decoding, not encoding. It's normal for it to not work.

**Reply by Str1atum** [reply]
thanks, that's a pretty good explanation!
